### DISCUSSION TITLE
useHTTP10

### DISCUSSION THREAD

**Discussion by OleksandrDiachenko** [khoih-prog/AsyncHTTPRequest_Generic#52] [Ideas]
Can you add .useHTTP10 to the library in the future?

**Comment by khoih-prog** [comment]
Hi @OleksandrDiachenko 

Even it's simple to modify to use `HTTP1.0` as an option, can you specify **your use-case** as well as **test by just changing a line in the library** from

https://github.com/khoih-prog/AsyncHTTPRequest_Generic/blob/8b03589b0224eb25401454050c44dad7b1391634/src/AsyncHTTPRequest_Impl_Generic.h#L1330

to

```cpp
  _request->write(" HTTP/1.0\r\n");
```

The benefits of `HTTP/1.1` 
- provides for **persistent** connections, which implies that many responses and requests can be sent over the same HTTP connection. 
- supports **chunk transfers**
- too many more new features. Check [**RFC2616** section-19.6.1](https://www.rfc-editor.org/rfc/rfc2616.html#section-19.6.1)

But beware that the responses, if using `HTTP/1.0`, really depend on the new `HTTP/1.1` servers

Check

1. [**How does a HTTP 1.1 server respond to a HTTP 1.0 request?**](https://stackoverflow.com/questions/35850518/how-does-a-http-1-1-server-respond-to-a-http-1-0-request)
2. [**RFC2616** section-19.6.1](https://www.rfc-editor.org/rfc/rfc2616.html#section-19.6.1)

You have to fully test and report everything is OK there as I have no time to deal with this now. If OK then, I'll move forward to add the `HTTP/1.0` option.
